wgpu-native/src/device.rs

```diff
@@ -168,11 +196,38 @@ impl<B: hal::Backend> DestroyedResources<B> {
             }
         }
 
+        for buffer_id in self.ready_to_map.drain(..) {
+            let mut buffer = buffer_guard.get_mut(buffer_id);
+            for operation in buffer.pending_map_operations.drain(..) {
```

Comment:
I don't think we expect more than a single pending map operations there, let's make it an `Option`